src/sentry/tsdb/base.py

```diff
@@ -212,7 +212,7 @@ def normalize_ts_to_rollup(self, epoch, seconds):
         """
         return int(epoch / seconds)
 
-    def get_optimal_rollup(self, start_timestamp, end_timestamp):
+    def get_optimal_rollup(self, start_timestamp, end_timestamp) -> int:
```

Comment:
Looks like we're still missing types for a fair few input params here and on other functions

Reply:
@wedamija Thank you for the review Dan. I was upgrading the typing for this file based on the ```# minimal strictness settings``` of the mypy configuration in pyproject.toml

I noticed at the bottom of pyproject.toml there seemed to be an override with ```disallow_untyped_defs``` for a few modules. Is that what I should be following?

Reply:
I'm not totally sure tbh, but it seems to me that we should try and fully type modules when possible. #discuss-typing might have more info

Reply:
The current methodology is to type things to the point that the Sentry type checker is passing. Then we can slowly ramp up the strictness.

Normally I would agree with Dan about fully typing this module, but it's a big one, and legacy, and used in a lot of places, so getting it to pass the current strictness of typing feels correct to me. We can tackle the typing in smaller chunks.